### DISCUSSION TITLE
[Feature] Facial Recognition via API on external images

### DISCUSSION THREAD

**Discussion by cmprmsd** [immich-app/immich#14340] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Since the facial recognition is working very well and is already integrated into the API partially it would be great to have an API endpoint which takes an (uploaded/external) image and returns identified persons. That would be great for automation usecases with [n8n](https://n8n.io/) or similar external automation software. 

I think the alternative is fiddling with the database and forking the facial recognition into its own standalone script.

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
#9285

**Reply by cmprmsd** [reply]
I'd say that's a similar request but not a duplicate as this one is only targeting the API and no UI integration would be neccessary. Automation thingies! 😄

**Reply by bo0tzz** [reply]
I get that, but we're not gonna add complex features to our API just for external use ;)

**Reply by cmprmsd** [reply]
Ah, got it! :)
Would a PR with such a feature would be merged potentially?

**Reply by cmprmsd** [reply]
I got that sorted out at last... :D 
https://github.com/cmprmsd/immich-face-api
